packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.quickactions;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ShortcutInfo;
+import android.content.pm.ShortcutManager;
+import android.graphics.drawable.Icon;
+import android.os.Build;
+import android.os.Bundle;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import io.flutter.plugin.common.PluginRegistry.Registrar;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** QuickActionsPlugin */
+@SuppressWarnings("unchecked")
+public class QuickActionsPlugin implements MethodCallHandler {
+  private final Activity activity;
+  private static MethodChannel channel;
```

Comment:
It doesn't seem right that this is static since it's tied to the `registrar` -- why not make it a final instance variable and pass it to the constructor?  Then `ShortcutHandlerActivity` will either have to become an inner class or take a `MethodChannel` argument in its constructor...